src/main/scala/com/databricks/spark/csv/DefaultSource.scala

```diff
@@ -102,6 +102,8 @@ class DefaultSource
       throw new Exception("Ignore white space flag can be true or false")
     }
 
+    val charset = parameters.getOrElse("charset", DefaultCharset)
```

Comment:
I think it is a good idea to validate the charset. Either here or in the utility function that uses it to read data into an RDD.